astroid/manager.py

```diff
@@ -365,8 +366,24 @@ def bootstrap(self):
     def clear_cache(self) -> None:
         """Clear the underlying cache, bootstrap the builtins module and
         re-register transforms."""
+        # import here because of cyclic imports
+        # pylint: disable=import-outside-toplevel
+        from astroid.inference_tip import clear_inference_tip_cache
```

Comment:
I'm still hesitant about this because of all the headaches the cyclic imports are giving us here, but I guess it is better than the status quo?

Reply:
I'm under the impression that if we fix the other cyclic imports issue, then this one will also be resolved. But then I might be wrong and we could be imbricating another ouroboro in our big ball of ouroboros. It's hard to know until we get to the bottom of the nature of the existing cyclic imports.

Reply:
I think this would be an alternative. Is this better or worse?
```python
diff --git a/astroid/__init__.py b/astroid/__init__.py
index 14a61c2a..cec22085 100644
--- a/astroid/__init__.py
+++ b/astroid/__init__.py
@@ -80,7 +80,7 @@ from astroid.exceptions import (
     UnresolvableName,
     UseInferenceDefault,
 )
-from astroid.inference_tip import _inference_tip_cached, inference_tip
+from astroid.inference_tip import _inference_tip_cached, inference_tip, clear_inference_tip_cache
 from astroid.objects import ExceptionInstance
 
 # isort: off
diff --git a/astroid/manager.py b/astroid/manager.py
index 30718e24..1ad5c48b 100644
--- a/astroid/manager.py
+++ b/astroid/manager.py
@@ -12,6 +12,7 @@ import types
 import zipimport
 from typing import TYPE_CHECKING, ClassVar, List, Optional
 
+import astroid
 from astroid.exceptions import AstroidBuildingError, AstroidImportError
 from astroid.interpreter._import import spec
 from astroid.modutils import (
@@ -359,20 +360,14 @@ class AstroidManager:
 
     def clear_cache(self):
         """Clear the underlying caches. Also bootstraps the builtins module."""
-        # import here because of cyclic imports
-        # pylint: disable=import-outside-toplevel
-        from astroid.inference_tip import clear_inference_tip_cache
-        from astroid.interpreter.objectmodel import ObjectModel
-        from astroid.nodes.node_classes import LookupMixIn
-
-        clear_inference_tip_cache()
+        astroid.clear_inference_tip_cache()
 
         self.astroid_cache.clear()
 
         for lru_cache in (
-            LookupMixIn.lookup,
+            astroid.node_classes.LookupMixIn.lookup,
             _cache_normalize_path_,
-            ObjectModel.attributes,
+            astroid.interpreter.objectmodel.ObjectModel.attributes,
         ):
             lru_cache.cache_clear()
```

Reply:
Hm, probably better as is. This is just hiding the cyclic import.

Reply:
I guess the other alternative is to move the barebones of the cached routines to a new file. But not sure there's much upside. I think there's greater ROI elsewhere.

Reply:
Agreed!